src/clims/services/substance.py

```diff
@@ -291,8 +289,19 @@ class SubstanceBase(ExtensibleBase):
     WrappedVersion = SubstanceVersion
 
     def __init__(self, **kwargs):
+        self._cached_parents = self._fetch_parents(kwargs)
         super(SubstanceBase, self).__init__(**kwargs)
 
+    def _fetch_parents(self, kwargs):
+        wrapped_version = kwargs.get("_wrapped_version", None)
+        parents = kwargs.pop("parents", None)
+
+        if wrapped_version and parents:
```

Comment:
Perhaps the error message here should be more explanatory? Something along the lines of `A substance must either be created from a set of parents or a wrapped version`? Not sure that is exactly what the problem is, but I think that it is a good practices to let the AssertionError give a more though description of what went wrong.

Reply:
ok!